src/lib/Group.ts

```diff
@@ -203,6 +203,13 @@ export class Group<
     return XMTP.isGroupActive(this.client.address, this.id)
   }
 
+  async addedByAddress(skipSync = false): Promise<string> {
+    if (!skipSync) {
+      await this.sync()
+    }
```

Comment:
```suggestion
  async addedByAddress(): Promise<string> {
```

Judging from your test I actually don't think we want this sync option here at all. It's a costly call and doesn't bringing any benefit as this should be set once and not change.